### PR TITLE
[symfony/HttpKernel] Throws an error when the generated class name is invalid.

### PR DESCRIPTION
**Pull request by drupol**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #30845 
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Tests/KernelTest.php [hunk 1]
```diff
@@ -64,6 +64,19 @@ public function testClone()
         $this->assertNull($clone->getContainer());
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
```
Comment (Simperfit) [inline src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: can you check the message too ?

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: Sure, will append a commit in a few minutes.

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: Done.

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: See 415254a412aa3387a11cd9901b4ab32efa443679

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 2]
```diff
@@ -442,14 +442,21 @@ protected function build(ContainerBuilder $container)
...
      * @return string The container class
      */
     protected function getContainerClass()
     {
         $class = \get_class($this);
         $class = 'c' === $class[0] && 0 === strpos($class, "class@anonymous\0") ? get_parent_class($class).str_replace('.', '_', ContainerBuilder::hash($class)) : $class;
+        $class = $this->name.str_replace('\\', '_', $class).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+
+        if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $class)) {
+            throw new \InvalidArgumentException(sprintf('The generated classname is invalid. Make sure that the environment name contains only valid characters.'));
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Kernel.php]: We are trying to make exception message as one sentence. Could be `The generated class name can only contain characters allowed in PHP class names.`

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php (hunk 2 above)
Comment (drupol) [reply src/Symfony/Component/HttpKernel/Kernel.php]: Doing it right now.

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Kernel.php]: Fixed in c8734c29aec23e66b1123f04645e6edfe66ec77d

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Kernel.php]: `Environment "%s" contains invalid characters.`?

Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Kernel.php]: I think having an explanation about which chars are valid is important. Having the environment name is indeed nice: `Environment "%s" can only contain characters allowed in PHP class names.`

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Kernel.php]: Ok will add it.

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Kernel.php]: I did a mix of both sentences, I'm waiting for your feedback again.

Comment (drupol) [reply src/Symfony/Component/HttpKernel/Kernel.php]: See 1fd718ae1bf370a6d880a6e0d2010c52b3b608d0

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @drupol.

**Pull request merged**